### DISCUSSION TITLE
Commit mode and isolation level scrolldown

### DISCUSSION THREAD

**Discussion by yana-markovich** [dbeaver/dbeaver#21259] [DBeaver Development]
I'm trying to load JDBC driver that i wrote to DBeaver, but i have some issues with the commit mode and isolation level scroll down menu. It is not clickable and i cannot change the isolation level. Commit mode is changeable through the 'edit connection' window. But the isolation level does not work from there as well, even after 'Test connection' is executed.

![image](https://github.com/dbeaver/dbeaver/assets/127675622/af179eb6-47c1-4edc-9bfc-8e0d2af05a2f)

![image](https://github.com/dbeaver/dbeaver/assets/127675622/d14cb796-7231-4f70-8c2c-aca455dd4b47)

I cannot click on the marked parts. 
What can be the issue?

**Comment by E1izabeth** [comment]
It looks like something became broken. Could you try to `Reset Perspective` or `Reset UI settings`?

![image](https://github.com/dbeaver/dbeaver/assets/28875055/8f38cfe3-4503-45ea-ae75-8b7027c80fc5)

![image](https://github.com/dbeaver/dbeaver/assets/28875055/b3356033-6572-4a61-acce-508b6e95a5c9)

**Reply by yana-markovich** [reply]
Hi Elizabeth, thank you for answering.
Unfortunately, it did not fix my issue. It happens only with **my** JDBC driver. For others, those buttons work as expected.
Can you think of a function (or more than one) that might break this? (If it is not implemented)

**Comment by E1izabeth** [comment]
Usually those buttons are enabled when transaction is active. We check it in `DataSourcePropertyTester` class (see `case` for `PROP_TRANSACTION_ACTIVE` in `test()` method)

**Reply by yana-markovich** [reply]
The transaction is active, I have a transaction log (in dbeaver). 
For other jdbc drivers, those buttons work even before I executed anything, so no active transaction for sure.

**Comment by flambchap** [comment]
DBeaver fetches the data for these combo boxes from the database itself. I suspect that the driver does not properly report available isolation levels.

**Reply by yana-markovich** [reply]
Thank you :) 
The JDBC method supportsTransactionIsolationLeve (in DatabaseMetaData)l is implemented and working.
